pkg/splunk/enterprise/util.go

```diff
@@ -292,3 +294,236 @@ func DeleteOwnerReferencesForS3SecretObjects(client splcommon.ControllerClient,
+
+	return false
+}
+
+// changeAppSrcDeployInfoStatus sets the new status to all the apps in an AppSrc if the given repo state and deploy status matches
+// primarly used in Phase-3
+func changeAppSrcDeployInfoStatus(appSrc string, appSrcDeployStatus map[string]enterprisev1.AppSrcDeployInfo, repoState enterprisev1.AppRepoState, oldDeployStatus enterprisev1.AppDeploymentStatus, newDeployStatus enterprisev1.AppDeploymentStatus) {
+	scopedLog := log.WithName("changeAppSrcDeployInfoStatus").WithValues("Called for AppSource: ", appSrc, "repoState", repoState, "oldDeployStatus", oldDeployStatus, "newDeployStatus", newDeployStatus)
+
+	if appSrcDeploymentInfo, ok := appSrcDeployStatus[appSrc]; ok {
+		appDeployInfoList := appSrcDeploymentInfo.AppDeploymentInfoList
+		for _, appDeployInfo := range appDeployInfoList {
+			// Modify the app status if the state and status matches
+			if appDeployInfo.RepoState == repoState && appDeployInfo.DeployStatus == oldDeployStatus {
+				appDeployInfo.DeployStatus = newDeployStatus
+			}
+		}
+
+		// Update the Map entry again
+		appSrcDeployStatus[appSrc] = appSrcDeploymentInfo
+		scopedLog.Info("Complete")
+	} else {
+		// Ideally this should never happen, check if the "IsDeploymentInProgress" flag is handled correctly or not
+		scopedLog.Error(nil, "Could not find the appSource")
+	}
+}
+
+// setStateAndStatusForAppDeployInfo sets the state and status for an App
+func setStateAndStatusForAppDeployInfo(appDeployInfo enterprisev1.AppDeploymentInfo, repoState enterprisev1.AppRepoState, deployStatus enterprisev1.AppDeploymentStatus) {
+	appDeployInfo.RepoState = repoState
+	appDeployInfo.DeployStatus = deployStatus
+}
+
+// setStateAndStatusForAppDeployInfoList sets the state and status for a given list of Apps
+func setStateAndStatusForAppDeployInfoList(appDeployList []enterprisev1.AppDeploymentInfo, state enterprisev1.AppRepoState, status enterprisev1.AppDeploymentStatus) bool {
+	var modified bool
+	for _, appInfo := range appDeployList {
+		setStateAndStatusForAppDeployInfo(appInfo, state, status)
+		modified = true
+	}
+
+	return modified
+}
+
+// handleAppRepoChanges parses the remote storage listing and updates the repoState and deployStatus accordingly
+// clinet and cr are used when we put the glue logic to hand-off to the side car
+func handleAppRepoChanges(client splcommon.ControllerClient, cr splcommon.MetaObject,
+	appDeployContext *enterprisev1.AppDeploymentContext, remoteObjListingMap map[string]splclient.S3Response, appFrameworkConfig *enterprisev1.AppFrameworkSpec) (bool, error) {
+	crKind := cr.GetObjectKind().GroupVersionKind().Kind
+	scopedLog := log.WithName("handleAppRepoChanges").WithValues("kind", crKind, "name", cr.GetName(), "namespace", cr.GetNamespace())
+	var err error
+
+	scopedLog.Info("App Listing", " received App listing for %d app sources", len(remoteObjListingMap))
+	if remoteObjListingMap == nil || len(remoteObjListingMap) == 0 {
+		scopedLog.Error(nil, "remoteObjectList is empty")
```

Comment:
won't throw err here? Just logging it?

Reply:
Ya, we intentionally not returning an error, to cover the case in which the customer wants to disable/delete all the apps that were deployed.